src/biokbase/auth.py

```diff
@@ -41,6 +41,48 @@ def __init__(self, info_dict: dict, token: str = None):
         self.token = token if token is not None else info_dict.get("token")
 
 
+class UserRole:
+    def __init__(self, role_info):
+        self.role_id = role_info.get("id")
+        self.description = role_info.get("desc")
+
+
+class PolicyId:
+    def __init__(self, policy_info):
+        self.policy_id = policy_info.get("id")
+        self.agree_date = policy_info.get("agreedon")
+
+
+class Identity:
+    def __init__(self, ident_info):
+        self.ident_id = ident_info.get("id")
+        self.provider = ident_info.get("provider")
+        self.provider_user = ident_info.get("provusername")
+
+
+class UserInfo:
+    """
+    A container for user information that comes from the Auth service. Does not
+    hold the token itself.
+    """
+
+    def __init__(self, user_dict: dict):
+        self.user = user_dict.get("user")
```

Comment:
is this a user ID? user name? Would be good to ensure that `Identity.provider_user` and `UserInfo.user` are both of the same type, either IDs or names.

Reply:
Can I ask why you need the identity provider user name? That really just an informational thing to help the user know what links to what. It's not something I'd expect to be used in any other KBase code

Reply:
I'm not sure we need any of it. I'm kinda just trying to be thorough and future-proofing here. Really, all that's ACTUALLY needed at this time is the user id, display name, and anon id.

Reply:
In that case I would strongly recommend not making it part of the local state and adding a comment to the effect that the provider username should only be used in the context of the user trying to figure out which accounts are linked to their KBase account

Reply:
I don't think any of that's actually necessary. I'll just trim that object down to what's needed and we can expand in the future if we need to. Which I doubt, as this is the first change to the auth client in ~6 years.